lib/ex_vault/utils.ex

```diff
@@ -4,4 +4,19 @@ defmodule ExVault.Utils do
   def uri_encode(val) when is_integer(val), do: to_string(val)
   def uri_encode(val) when is_atom(val), do: val |> to_string() |> uri_encode()
   def uri_encode(val), do: URI.encode(val, &URI.char_unreserved?/1)
+
+  def to_uri_path(secret_path) when is_list(secret_path),
+    do: secret_path |> Enum.map(&uri_encode/1) |> Enum.join("/")
+
+  def to_uri_path(secret_path), do: uri_encode(secret_path)
+
+  def handle_response({:ok, %{status: status, body: %{"data" => _}} = resp})
+      when status in [200, 204],
```

Comment:
This is incorrect, since 204 response will not have the body.

Let's have `handle_no_content_response` to match against 204 + empty body, while `handle_ok_response` for 200 + data body. By doing that we can see the caller function will return a response with body or without body.